awscrt/auth.py

```diff
@@ -306,6 +306,63 @@ def new_delegate(cls, get_credentials):
         binding = _awscrt.credentials_provider_new_delegate(get_credentials)
         return cls(binding)
 
+    @classmethod
+    def new_cognito(
+            cls,
+            endpoint,
+            identity,
+            tls_ctx,
+            logins=None,
+            custom_role_arn=None,
+            client_bootstrap=None,
+            http_proxy_options=None):
+        """
+        Creates a provider that sources credentials from the AWS Cognito Identity service.
+
+        Args:
+            endpoint (str): Cognito Identity service regional endpoint to source credentials from.
+
+            identity (str): Cognito identity to fetch credentials relative to.
+
+            tls_ctx (ClientTlsContext): Client TLS context to use when querying cognito credentials by HTTP.
+
+            logins (Optional[List[Tuple[str, str]]]): List of tuples specifying pairs of identity provider name and
+                token values, representing established login contexts for identity authentication purposes
+
+            custom_role_arn (Optional[str]): ARN of the role to be assumed when multiple roles were received in the
+                token from the identity provider.
+
+            client_bootstrap (Optional[ClientBootstrap]): Client bootstrap to use when initiating socket connection.
+                If not set, uses the static default ClientBootstrap instead.
+
+            http_proxy_options (Optional[HttpProxyOptions]): Optional HTTP proxy options.
+                If None is provided then an HTTP proxy is not used.
+
+        Returns:
+            AwsCredentialsProvider:
+        """
+
+        assert isinstance(endpoint, str)
+        assert isinstance(identity, str)
+        assert isinstance(tls_ctx, ClientTlsContext)
+        assert isinstance(logins, list) or logins is None
+        assert isinstance(custom_role_arn, str) or custom_role_arn is None
+        assert isinstance(http_proxy_options, awscrt.http.HttpProxyOptions) or http_proxy_options is None
```

Comment:
trivial/debatable: this isn't really necessary anymore. We used to do this a lot because our error reporting from C was really bad, so catching errors earlier in Python was preferable. But our error-reporting from C has improved a lot.

and Python isn't a strictly typed language, and stuff like "duck typing" is not uncommon

and stuff like this makes it hard to accept a list or tuple

Reply:
First reaction is to just drop the logins assert.  Does that seem reasonable?